### DISCUSSION TITLE
Pytest Mark Warning when running pytest.

### DISCUSSION THREAD

**Discussion by tindi-plus** [networkx/networkx#6172] [General]
While runing pytest, I usually see the pytest warning:

`============================================ warnings summary ============================================= 
networkx\drawing\tests\test_pylab.py:422
  ~\networkx\networkx\drawing\tests\test_pylab.py:422: PytestUnknownMarkWarning: Unknown pytest.mark.mpl_image_compare - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html        
    @pytest.mark.mpl_image_compare`
The marker can be found at:

```
networkx\networkx\drawing\tests\test_pylab.py

@pytest.mark.mpl_image_compare
def test_house_with_colors():
    G = nx.house_graph()
    # explicitly set positions
    fig, ax = plt.subplots()
    pos = {0: (0, 0), 1: (1, 0), 2: (0, 1), 3: (1, 1), 4: (0.5, 2.0)}

    #Plot nodes with different properties for the "wall" and "roof" nodes
    nx.draw_networkx_nodes(
        G,
        pos,
        node_size=3000,
        nodelist=[0, 1, 2, 3],
        node_color="tab:blue",
    )
    nx.draw_networkx_nodes(
        G, pos, node_size=2000, nodelist=[4], node_color="tab:orange"
    )
    nx.draw_networkx_edges(G, pos, alpha=0.5, width=6)
    # Customize axes
    ax.margins(0.11)
    plt.tight_layout()
    plt.axis("off")
    return fig
```
The warning says that the marker `mpl_image_compare` is not registered. I think this can be solved (to make pytest happy :) ) by registering the marker, however, I couldn't find anywhere else where the marker is used. So I am thinking, why use a marker if it is used only once?

##Environment
```
=========================================== test session starts ===========================================
platform win32 -- Python 3.9.13, pytest-7.1.3, pluggy-1.0.0
rootdir: ~\networkx
plugins: cov-4.0.0
collected 4959 items / 3 skipped
```

**Comment by MridulS** [comment]
You need to install https://github.com/matplotlib/pytest-mpl to register `mpl_image_compare`. We use it only once right now but hopefully we will use it more in the future.

**Reply by tindi-plus** [reply]
Okay. Thank you for the clarification.
